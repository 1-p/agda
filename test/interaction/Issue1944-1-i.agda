-- Andreas, AIM XXIII 2016-04-21 Overloaded projections

-- Milestone 1: Check overloaded projections on rhs (without postponing).

{-# OPTIONS --allow-unsolved-metas #-}

module _ (A : Set) (a : A) where

record R B : Set where
  field f : B
open R

record S B : Set where
  field f : B
open S

r : R A
R.f r = a

s : S A
S.f s = f r

t : R A → S A
S.f (t r) = f r

u : _
u = f s

-- interactive

hole0 : A
hole0 = {! f s !}  -- normalize me

hole1 = {!λ r → f (t r)!} -- normalize me
