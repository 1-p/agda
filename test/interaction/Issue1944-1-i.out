(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue1944-1-i (Issue1944-1-i.agda). " t)
(agda2-info-action "*Type-checking*" "Finished Issue1944-1-i. " t)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : A ?1 : _18 A a ?2 : .A Sort _17 [ at Issue1944-1-i.agda:34,1-6 ] _18 : Set (_17 A a) [ at Issue1944-1-i.agda:34,1-6 ] " nil)
((last . 1) . (agda2-goals-action '(0 1 2)))
(agda2-status-action "")
(agda2-info-action "*Normal Form*" "a" nil)
((last . 1) . (agda2-goals-action '(0 1 2)))
(agda2-status-action "")
(agda2-info-action "*Normal Form*" "a" nil)
((last . 1) . (agda2-goals-action '(0 1 2)))
(agda2-status-action "")
(agda2-info-action "*Goal type etc.*" "Goal: A Have: A ———————————————————————————————————————————————————————————— a : A A : Set" nil)
((last . 1) . (agda2-goals-action '(0 1 2)))
cannot read: IOTCM "Issue1944-1-i.agda" None Indirect (Cmd_compute True 1 noRange "\ r -> f (t r)")
cannot read: IOTCM "Issue1944-1-i.agda" None Indirect (Cmd_goal_type_context_infer Normalised 1 noRange "\ r -> f (t r)")
(agda2-give-action 2 "x")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : A ?1 : _18 A a Sort _17 [ at Issue1944-1-i.agda:34,1-6 ] _18 : Set (_17 A a) [ at Issue1944-1-i.agda:34,1-6 ] " nil)
((last . 1) . (agda2-goals-action '(0 1)))
